include/oxen/quic/ip.hpp

```diff
@@ -72,24 +91,33 @@ namespace oxen::quic
         {}
 
       public:
-        // Network order
+        // Host order
         uint64_t hi, lo;
 
-        // Host order string constructor
-        ipv6(const std::string& ip);
+        constexpr ipv6() = default;
+
+        constexpr std::optional<ipv6> next_ip() const
+        {
+            // If lo will not overflow, increment and return
+            if (not will_overflow(lo))
+                return ipv6{hi, lo + 1};
+
+            // If lo is INT_MAX, then:
+            //  - if hi can be incremented, ++hi and set lo to all 0's
+            //  - else, return nullopt
+            if (not will_overflow(hi))
+                return ipv6{hi + 1, uint64_t{0}};
+
+            return std::nullopt;
```

Comment:
Can also simplify here by dropping `will_overflow` here and let it overflow with detection:
```suggestion
            auto next_lo = lo + 1;
            auto next_hi = hi + (next_lo == 0);
            if (next_hi == 0)
                return std::nullopt;
            return ipv6{next_hi, next_lo};
```